homeassistant/components/ted/config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Config flow for TED integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import httpx
+import tedpy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import (
+    ConfigEntry,
+    OptionsFlow,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .const import (
+    CONF_MTU_ENERGY_DAILY,
+    CONF_MTU_ENERGY_MTD,
+    CONF_MTU_ENERGY_NOW,
+    CONF_MTU_POWER_VOLTAGE,
+    CONF_SPYDER_ENERGY_DAILY,
+    CONF_SPYDER_ENERGY_MTD,
+    CONF_SPYDER_ENERGY_NOW,
+    DOMAIN,
+    OPTION_DEFAULTS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TED."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize an TED flow."""
+        self.name = None
+        self.serial = None
+        self.model = None
+
+    @callback
+    def _async_generate_schema(self):
```

Comment:
Better yet, I took out this function method since it looks like very few other components have an `async_generate_schema` method. Instead, they all create the `vol.Schema` inside the `async_step_user`.